### ISSUE TITLE
Update System.ObsoleteAttribute targets to include assembly

### ISSUE THREAD

**Issue by marek-safar** [dotnet/corefx#24915]
Currently, ObsoleteAttribute can be placed on types and members only which is very inconvenient when you need to obsolete the whole assembly. We already have quite a few cases in Xamarin but .NET is having the same problem with for example System.Data.OracleClient assembly which has been deprecated.

This change will also require C# compiler update.

**Comment by joperezr** [comment]
@marek-safar are you interested in providing a formal api proposal so that we take a look at it?

**Comment by marek-safar** [comment]
I could do that, do we have any template for it?

**Comment by joperezr** [comment]
Yes, here is an example that uses the template: #271. You don't need to create a new issue for it, you can just update the top comment on this one to look like that template. This is also a good resource of how the review process goes in general in case you wondered https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/api-review-process.md

**Comment by marek-safar** [comment]
Is there any work left to do from me?

**Comment by joperezr** [comment]
@jkotas any initial feedback on this? Personally, I'm not sure there is enough value for doing this since I don't think it would be used a lot.

**Comment by jkotas** [comment]
Is there work required in compilers to recognize this attribute on assembly, or does it "just work"?

> I'm not sure there is enough value for doing this

It is a very trivial edit. Small value for tiny amount of work sounds fine to me.

This change does have value for people out there who are perfectly happy to use ObsoleteAttribute to mark deprecated APIs.

This change does not have value for the core framework currently. The current API deprecation plan for core framework is not based on the ObsoleteAttribute. Instead, it is based on external database in https://github.com/dotnet/platform-compat tool.
